scripts/validate-oyd-vnet.sh

```diff
@@ -0,0 +1,368 @@
+    has_subscription_scope=$(echo $scope | jq --arg expected_scope "/subscriptions/$subscription_id" '.==$expected_scope')
+
+    if [[ $has_resource_scope == "false" && $has_subscription_scope == "false" ]]; then
+        echo "Role '$role_name' for resource '$resource_id' is missing the correct scope."
+        echo "Please run the following script to ensure that role assignments are set correctly for your resources: https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/scripts/role_assignment.sh"
+        exit 1
+    fi
+
+    echo "Role assignment scope is valid!"
+}
+
+function check_resources_subnet_connection() {
+    subscription_id=$1
+    vnet=$2
+    resources=$3
+    echo "Checking resources subnet connection to vnet..."
+    # JSON string representing a list of resource IDs
+    resource_ids=$(echo $resources | jq '.[].id' | jq -s)
+
+    # JSON string representing a list of subnets for which all of the above resources are associated through "Approved" private endpoint connections
+    resources_subnets=$(echo $resources | jq '.[].private_endpoints | flatten | .[].properties.privateEndpoint.id' | tr -d '"' | xargs az network private-endpoint show --subscription $subscription_id --ids 2>/dev/null | jq '.[] | select(any(.privateLinkServiceConnections[]; .privateLinkServiceConnectionState.status == "Approved"))' | jq -s 'group_by(.subnet.id)[] | {(.[0].subnet.id): [.[] | .privateLinkServiceConnections[].privateLinkServiceId]}' | jq --argjson ids "$resource_ids" 'map_values($ids-.) | to_entries | .[].key')
+
+    resources_subnets_count=$(echo $resources_subnets | jq -s 'length')
+    if [[ $resources_subnet_count == 0 ]]; then
+        echo "The given resources are not all connected to the same virtual network subnet via private endpoint connections.  Please check that each resource has a private endpoint connection to a subnet in your virtual network and try again."
+        exit 1
+    fi
+
+    echo "All resources are connected to the following subnet: $resources_subnets"
+}
+
+function check_gateway_subnet_connection() {
+    vnet_id=$1
+    gateway_resource=$2
+    gateway_resource_id=$(get_resource_id_from_resource "$gateway_resource")
+    echo "Checking VPN gateway is connected to virtual network"
+
+    gateway_subnet_is_in_vnet=$(echo $gateway_resource | jq --arg vnet_id $vnet_id '.properties.ipConfigurations[].properties.subnet.id | contains($vnet_id)')
+
+    if [[ $gateway_subnet_is_in_vnet != "true" ]]; then
+        echo "The VPN gateway $gateway_resource_id is not in a subnet of virtual network $vnet_id -- please follow the instructions here to configure the VPN gateway.  https://learn.microsoft.com/en-us/azure/vpn-gateway/tutorial-create-gateway-portal#VNetGateway"
+    fi
+
+    echo "VPN gateway is attached to the virtual network!"
+}
+
+function check_ame_private_endpoint_connection() {
+    echo "Checking AME private endpoint connection..."
```

Comment:
Change "AME" to "Microsoft managed", because external customers don't know what is AME.

Reply:
Thanks for pointing that out.  I have updated all references to AME to use the suggested language.